tests/tt_metal/tt_metal/llk/test_reduce.cpp

```diff
@@ -389,7 +389,7 @@ void run_single_core_reduce_program(tt_metal::IDevice* device, const ReduceConfi
     }
     // recover a linear view of input vector for consumption by gold_ function
     std::vector<uint16_t> src_linear = convert_layout<uint16_t>(
-        u16_src0_vec,
+        tt::stl::MakeConstSpan(u16_src0_vec),
```

Comment:
I believe this is only necessary where the template parameter is deduced, i.e.
```cpp
convert_layout(tt::stl::MakeConstSpan(u16_src0_vec), ...)
```
or
```cpp
convert_layout<uint16_t>(u16_src0_vec, ...)
```
but not
```cpp
convert_layout<uint16_t>(tt::stl::MakeConstSpan(u16_src0_vec), ...)
```
would you mind updating accordingly? I believe it should reduce the size of the diff a fair amount.